speechbrain/lobes/models/transformer/Conformer.py

```diff
@@ -99,20 +142,149 @@ def __init__(
             nn.Dropout(dropout),
         )
 
-    def forward(self, x, mask=None):
+    def _do_conv(self, x, inhibit_padding: bool):
+        if not inhibit_padding:
+            out = self.conv(x)
+        else:
+            # let's keep backwards compat by pointing at the weights from the
+            # already declared Conv1d.
+
+            # we do not need to edit the bottleneck as it is pointwise (i.e.
+            # time step by time step), thus, it doesn't need padding along the
+            # time dimension
+            out = F.conv1d(
+                x,
+                weight=self.conv.weight,
+                bias=self.conv.bias,
+                stride=self.conv.stride,
+                padding=0,
+                dilation=self.conv.dilation,
+                groups=self.conv.groups,
+            )
+
+        return out
+
+    def forward(self, x, mask=None, dct_config: Optional[DCTConfig] = None):
```

Comment:
This function and the whole streaming thing is fairly complex. I propose that you write an integration test following the current one we have for Transducer but only to test streaming.,